server/plugin.go

```diff
@@ -774,9 +775,15 @@ func (p *Plugin) syncUsers() {
 				_, err := p.API.UpdateUser(mmUser)
```

Comment:
Would it be simpler if we split apart the checks for `username != mmUser.Username` and `msUser.DisplayName != mmUser.FirstName` so we could only generate usernames in one case?